### PR TITLE
ref(webpack): Move frontend application to `static/app`

### PR DESCRIPTION
**Pull request by billyvg**
This moves the frontend application (i.e. everything that we process with webpack) to `static/app` - webpack then outputs the compiled files to `src/sentry/static/sentry/dist` like before.

The motivation for this is:
1) have a shorter path to our frontend application
2) have a clearer understanding of what webpack processes and what is needed for server rendered views
3) allow for future work to separate assets that are versioned by contents (from webpack) vs by release (e.g. current behavior for static assets)

## Risks
* There's a risk here that an asset moved to `static/` was referenced outside of our webpack bundles. This mostly applies to `images` (and maybe `fonts`) as they would be the assets most likely to be referenced outside of webpack.
* This will probably cause some `git` pain 😬

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,9 +16,9 @@ in realtime. The server is in Python, but it contains a full API for
 sending events from any language, in any application.
 
 <p align="center">
-  <img src="https://github.com/getsentry/sentry/raw/master/src/sentry/static/sentry/images/screenshots/thumb-1.png" width="290">
-  <img src="https://github.com/getsentry/sentry/raw/master/src/sentry/static/sentry/images/screenshots/thumb-2.png" width="290">
-  <img src="https://github.com/getsentry/sentry/raw/master/src/sentry/static/sentry/images/screenshots/thumb-3.png" width="290">
+  <img src="https://github.com/getsentry/sentry/raw/master/docs/screenshots/thumb-1.png" width="290">
+  <img src="https://github.com/getsentry/sentry/raw/master/docs/screenshots/thumb-2.png" width="290">
+  <img src="https://github.com/getsentry/sentry/raw/master/docs/screenshots/thumb-3.png" width="290">
```
Comment (billyvg) [inline README.md:21]: Made a new `docs/` dir in repo root, as these images are only used in the README.

File: codecov.yml [hunk 2]
```diff
@@ -19,5 +19,5 @@ coverage:
   - src/.*?/migrations/.*
   - src/bitfield/.*
   - src/social_auth/.*
-  - src/sentry/static/sentry/app/routes.jsx
+  - static/app/routes.jsx
```
Comment (billyvg) [inline codecov.yml:22]:
```suggestion
  - static/app/routes.tsx
```

**Comment by EvanPurkhiser** [comment]
Approved for staging

File: static/less/shared-components.less [hunk 3]
```diff
@@ -376,7 +376,7 @@ table.table.key-value {
 .org-avatar {
   width: 30px;
   height: 30px;
-  background: @red url(../images/logos/sentry-avatar.png);
+  background: @red url(../images/sentry-avatar.png);
```
Comment (billyvg) [inline static/less/shared-components.less:379]: Not sure why this was in `logos/` in the first place, but moved to static/images

**Comment by billyvg** [comment]
Tested on staging, everything seems to work fine.

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**